### PR TITLE
Discussion #489: add CopyAffine transform

### PR DESCRIPTION
**Pull request by Albans98**
Fixes #489 .

**Description**
Add a CopyAffine transform with a target_key parameter to copy the affine from this key to all LabelMap objects of a given torchio subject.

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/master/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [x] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`
- [x] In-line docstrings updated
- [x] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [x] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

**Comment by romainVala** [comment]
merci alban

File: torchio/transforms/preprocessing/spatial/copy_affine.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                 target_key: str,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.target_key = target_key
+        self.args_names = ['target_key']
+
+    def apply_transform(self, subject: Subject) -> Subject:
+        affine = subject[self.target_key].affine
+        for image in subject.get_images(intensity_only=False):
+            if isinstance(image, LabelMap):
```
Comment (fepegar) [inline torchio/transforms/preprocessing/spatial/copy_affine.py:29]: Why does this affect label maps only?

**Review by fepegar — CHANGES_REQUESTED** [review]
Thanks for your PR! I added a comment. Could you please update the corresponding rst file in the documentation?

File: torchio/transforms/preprocessing/spatial/copy_affine.py (hunk 1 above)
Comment (romainVala) [reply torchio/transforms/preprocessing/spatial/copy_affine.py:29]: I agree, it should be all, except the target_key

**Comment by romainVala** [comment]
Hi, I am not sure to understand what the error is, 
should we change things ... ?

**Comment by fepegar** [comment]
> Hi, I am not sure to understand what the error is,
should we change things ... ?

I think the error is happening because of a Sphinx problem that was fixed in 6e614478ed3b273613c7a08ca2b3f5ed2bef6c52, before this branch was created. Don't worry about it. I'm going to take a look at this PR tonight or tomorrow.

**Comment by fepegar** [comment]
Thanks for your contribution, @Albans98!

@all-contributors please add @Albans98 for code

**Comment by Albans98** [comment]
Thanks for your time and your reviews !

**Comment by romainVala** [comment]
@Albans98 just find out, there is one line missing at the end of the init
```python
        self.args_names = ['target']
```

**Comment by fepegar** [comment]
That's my fault, I removed it thinking it was necessary. It's needed to reproduce the transform, right?

**Comment by romainVala** [comment]
may be @Albans98  can confirm, I don't know what the exact problem was, 
In our use case, we do save the transform history (to be able to reproduce later)

but anyway It make sens to have it for all the transform (preprocessing and augmentation)
(I quickly check, and there may be some other missing like OneHot)

**Comment by Albans98** [comment]
Hello, yes the problem is when we try to save the history of all applied transforms, we get an error 'TypeError: __init__() missing 1 required positional argument: 'target''. 

Adding the target in the list of args makes it possible to trace back to the 'target' value.

**Comment by fepegar** [comment]
Ok, let me fix that.

**Comment by fepegar** [comment]
Done. Thanks for pointing that out!

**Comment by Albans98** [comment]
Thank you for fixing this !

**Pull request merged**